tests/core/pyspec/eth2spec/test/capella/block_processing/test_process_deposit.py

```diff
@@ -56,3 +56,31 @@ def test_success_top_up_to_withdrawn_validator(spec, state):
         assert has_execution_withdrawal and is_withdrawable and has_non_zero_balance
     else:
         assert spec.is_fully_withdrawable_validator(validator, balance, current_epoch)
+
+
+@with_capella_and_later
+@spec_state_test
+def test_validator_invalid_high_funding_after_withdrawal(spec, state):
+    """
+    Simple extension of the previous test_success_top_up_to_withdrawn_validator in order to add a negative scenario
+    where a user deposits 2x MAX_EFFECTIVE_BALANCE.
+    """
+    validator_index = 5  # Example validator index
+
+    # Fully withdraw the validator
+    set_validator_fully_withdrawable(spec, state, validator_index)
+    assert state.balances[validator_index] > 0
+
+    next_epoch_via_block(spec, state)
+    assert state.balances[validator_index] == 0
+    assert state.validators[validator_index].effective_balance > 0
+
+    next_epoch_via_block(spec, state)
+    assert state.validators[validator_index].effective_balance == 0
+
+    # Make a top-up balance to validator - attempt to deposit double the max deposit)
+    deposit = spec.MAX_EFFECTIVE_BALANCE * 2
+
+    # Handle the negative test based on the readme - no post state
+    if deposit > spec.MAX_EFFECTIVE_BALANCE:
+        yield 'post', None
```

Comment:
Firstly, this test is flawed for a few reasons.

* This isn't invalid. It's possible deposit as much ETH as you want, but the validators _effective_ balance is limited.
* If it were invalid, you should use `run_deposit_processing(..., valid=False)` instead manually yielding.
* It doesn't handle the electra case (`spec.MAX_EFFECTIVE_BALANCE_ELECTRA`).
* What readme are you referring to? The test format?

Reply:
Hey again. No worries, this stuff is pretty complicated. I would suggest updating this test to check that a top up deposit greater than MAX_EFFECTIVE_BALANCE (before and after electra) still works. And that the validator's balance (not its effective balance) is not limited.